samcli/local/lambda_service/local_lambda_invoke_service.py

```diff
@@ -46,14 +48,65 @@ def create(self):
                                methods=['POST'],
                                provide_automatic_options=False)
 
+        # setup request validation before Flask calls the view_func
+        self._app.before_request = self.validate_request
+
         self._construct_error_handling()
 
+    def validate_request(self):
+        """
+        Validates the incoming request
+
+        Returns
+        -------
+        flask.Response
+            A Response is returned in the following cases:
+            1. The Request data is not json serializable
+            2. Query Parameters are sent to the endpoint
+            3. The Request Content-Type is not application/json
+
+            Otherwise this function does not return anything
+        """
+        flask_request = request
+        request_data = flask_request.get_data()
+
+        if not request_data:
+            request_data = b'{}'
+
+        request_data = request_data.decode('utf-8')
+
+        try:
+            json.loads(request_data)
+        except ValueError as json_error:
+            LOG.debug("Request body was not json.")
+            return LambdaErrorResponses.invalid_request_content(
+                "Could not parse request body into json: {}".format(str(json_error)))
+
+        if flask_request.args:
+            LOG.debug("Query parameters are in the request but not supported")
+            return LambdaErrorResponses.invalid_request_content("Query Parameters are not supported")
+
+        if flask_request.content_type.lower() != "application/json":
+            LOG.debug("%s media type is not supported. Must be application/json", flask_request.content_type)
+            return LambdaErrorResponses.unsupported_media_type(flask_request.content_type)
+
+        log_type = flask_request.headers.get('X-Amz-Log-Type', 'None')
+        if log_type != 'None':
+            LOG.info("log-type: %s is not supported. Ignoring X-Amz-Log-Type header", log_type)
```

Comment:
`debug` instead?

Reply:
I did this as info to display we are ignoring the `X-Amz-Log-Type` header, it feels wrong to only display that to users when the `--debug` is set. May be better to just exit there instead of still handling the request, which then it would make more sense to move this to debug.

Reply:
I think exiting and returning error would be better. It is explicit and less surprises.